traits.go

```diff
@@ -13,6 +13,72 @@ type Trait interface {
 	ValidateTrait() error
 }
 
+// OpenCloseTrait provides an implementation of the Smart Home OpenClose Trait Schema from Google Smart Home Actions
+type OpenCloseTrait struct {
+	DiscreteOnlyOpenClose bool
+	OpenDirection []string
```

Comment:
since "UP" and "DOWN" are validated by google these should be an enum rather then a string,

type OpenCloseTraitDirection string

const OpenCloseTraitDirectionUp OpenCloseTraitDirection "UP"
etc

(duplicate comment but it makes more sense to put it here)

Reply:
fixed